x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/RoleDescriptor.java

```diff
@@ -1356,6 +1356,11 @@ public String[] getDeniedFields() {
             return this.deniedFields;
         }
 
+        // TODO avoid this setter
```

Comment:
This is only used in `RemoteClusterSecurityLegacyCrossClusterApiKeysWithDlsFlsIT` to add a `query` field to an API key doc.